tests/test_preconditions.py

```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
     privileged: bool = True,
     user: int = 0,
     pid_mode: Optional[str] = "host",
+    extra_profiler_args: Optional[List[str]] = None,
 ) -> Container:
+    extra_profiler_args = extra_profiler_args or []
     return start_gprofiler_in_container_for_one_session(
         docker_client,
         gprofiler_docker_image,
         Path("/tmp"),
         Path("/tmp/collapsed"),
         [],
-        ["-d", "1"],
+        ["-d", "1"] + extra_profiler_args,
         privileged=privileged,
         user=user,
         pid_mode=pid_mode,
     )
 
 
+@contextmanager
+def run_gprofiler_container(
+    docker_client: DockerClient,
+    gprofiler_docker_image: Image,
+    privileged: bool = True,
+    user: int = 0,
+    pid_mode: Optional[str] = "host",
+    extra_profiler_args: Optional[List[str]] = None,
+) -> Container:
+    container: Container = None
+    try:
+        container = start_gprofiler(
+            docker_client, gprofiler_docker_image, privileged, user, pid_mode, extra_profiler_args
+        )
+        yield container
+    finally:
+        if container is not None:
+            container.stop()  # Fine to be run also if already stopped
```

Comment:
Leaving the "remove" for a future PR?

Reply:
yes, not related, part of https://github.com/Granulate/gprofiler/issues/600